cfme/tests/cloud/test_child_quota.py

```diff
@@ -0,0 +1,153 @@
+            "properties": {'instance_type': partial_match('m1.large')}
+        }
+    if provider.one_of(AzureProvider):
+        return {
+            "catalog": {'vm_name': vm_name, 'catalog_name': {'name': template_name}},
+            "environment": {'automatic_placement': True},
+            "properties": {'instance_type': partial_match('d2s_v3')},
+            "customize": {'admin_username': provisioning['customize_username'],
+                          'root_password': provisioning['customize_password']}
+        }
+
+
+@pytest.fixture
+def set_child_tenant_quota(request, appliance, new_child):
+    field, value = request.param
+    new_child.set_quota(**{'{}_cb'.format(field): True, field: value})
+    yield
+    # will refresh page as navigation to configuration is blocked if alerts are on the page
+    appliance.server.login_admin()
+    appliance.server.browser.refresh()
+    new_child.set_quota(**{'{}_cb'.format(field): False})
+
+
+@pytest.fixture(scope='module')
+def new_tenant(appliance):
+    collection = appliance.collections.tenants
+    tenant = collection.create(name='tenant{}'.format(fauxfactory.gen_alphanumeric()),
+                               description='tenant_des{}'.format(fauxfactory.gen_alphanumeric()),
+                               parent=collection.get_root_tenant())
+    yield tenant
+    if tenant.exists:
+        tenant.delete()
+
+
+@pytest.fixture(scope='module')
+def new_child(appliance, new_tenant):
+    """The fist child tenant doesn't act like child tenant but equivale"""
+    collection = appliance.collections.tenants
+    child_tenant = collection.create(name='tenant{}'.format(fauxfactory.gen_alphanumeric()),
+                                     description='tenant_des{}'.format(
+                                         fauxfactory.gen_alphanumeric()),
+                                     parent=new_tenant)
+    yield child_tenant
+    if child_tenant.exists:
+        child_tenant.delete()
+
+
+@pytest.fixture(scope='module')
+def new_group(appliance, new_child, new_tenant):
+    collection = appliance.collections.groups
+    group = collection.create(description='group_{}'.format(fauxfactory.gen_alphanumeric()),
+                              role='EvmRole-super_administrator',
+                              tenant='My Company/{}/{}'.format(new_tenant.name, new_child.name))
+    yield group
+    if group.exists:
+        group.delete()
+
+
+@pytest.fixture(scope='module')
+def new_user(appliance, new_group, new_credential):
+    collection = appliance.collections.users
+    user = collection.create(
+        name='user_{}'.format(fauxfactory.gen_alphanumeric()),
+        credential=new_credential,
+        email='xyz@redhat.com',
+        groups=new_group,
+        cost_center='Workload',
+        value_assign='Database')
+    yield user
+    if user.exists:
+        user.delete()
+
+
+# first arg of parametrize is the list of fixtures or parameters,
```

Comment:
Thanks for writing a bit of an explanation about what's going on here :) Can you also describe what the purpose of each param is. For example, what 'custom_prov_data', 'extra_msg', and 'approve' are being used for in the test?

Reply:
Sure I will update that.